controllers/clusters/postgresql_controller.go

```diff
@@ -468,6 +482,129 @@ func (r *PostgreSQLReconciler) handleUpdateCluster(
 	return models.ExitReconcile
 }
 
+func (r *PostgreSQLReconciler) createUser(
+	ctx context.Context,
+	l logr.Logger,
+	c *v1beta1.PostgreSQL,
+	uRef *v1beta1.UserReference,
+) error {
+	req := types.NamespacedName{
+		Namespace: uRef.Namespace,
+		Name:      uRef.Name,
+	}
+
+	u := &clusterresourcesv1beta1.PostgreSQLUser{}
+	err := r.Get(ctx, req, u)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			l.Error(err, "Cannot create a PostgreSQL user. The resource is not found", "request", req)
+			r.EventRecorder.Eventf(c, models.Warning, models.NotFound,
+				"User is not found, create a new one PostgreSQL User or provide correct userRef."+
+					"Current provided reference: %v", uRef)
+			return err
+		}
+
+		l.Error(err, "Cannot get PostgreSQL user", "user", u.Spec)
+		r.EventRecorder.Eventf(c, models.Warning, models.CreationFailed,
+			"Cannot get PostgreSQL user. user reference: %v", uRef)
+		return err
+	}
+
+	secret, err := v1beta1.GetDefaultPgUserSecret(context.TODO(), c.Name, c.Namespace, r.Client)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		r.EventRecorder.Eventf(
+			c, models.Warning, models.FetchFailed,
+			"Default user secret fetch is failed. Reason: %v",
+			err,
+		)
+
+		return err
+	}
+
+	defaultSecretNamespacedName := types.NamespacedName{
+		Namespace: secret.Namespace,
+		Name:      secret.Name,
+	}
+
+	if _, exist := u.Status.ClustersEvents[c.Status.ID]; exist {
+		l.Info("User is already existing on the cluster",
+			"user reference", uRef)
+		r.EventRecorder.Eventf(c, models.Normal, models.CreationFailed,
+			"User is already existing on the cluster. User reference: %v", uRef)
+
+		return nil
+	}
+
+	patch := u.NewPatch()
+
+	if u.Status.ClustersEvents == nil {
+		u.Status.ClustersEvents = make(map[string]string)
+	}
+
+	u.Status.DefaultSecretNamespacedName = defaultSecretNamespacedName.String()
```

Comment:
You need the reference to the default user in the `PostgreSQLUserController` to create a new one. But what if there is more than 1 cluster reference to the given user resource?
I think it would be better if we save the reference in another map like `map[ClusterID]DefaultSecretNamespacedName`. Or modify `ClusterEvents` to something like that:
```
type ClusterEvent struct {
    Event string `json:"event,omitempty"`
    DefaultSecretRef types.NamespacedName `json:"defautSecretRef,omitempty"`
}

type PostgreSQLUserStatus struct {
    ClusterEvents map[ClusterID]ClusterEvent `json:"clusterEvents,omitempty"
}
```

Reply:
nice catch, fixed and tested